walletlibrary/src/main/java/com/microsoft/walletlibrary/VerifiedIdClient.kt

```diff
@@ -30,31 +31,37 @@ class VerifiedIdClient(
 ) {
 
     // Creates an issuance or presentation request based on the provided input.
-    suspend fun createRequest(verifiedIdRequestInput: VerifiedIdRequestInput): Result<VerifiedIdRequest<*>> {
-        return try {
+    suspend fun createRequest(verifiedIdRequestInput: VerifiedIdRequestInput): VerifiedIdResult<VerifiedIdRequest<*>> {
+        return getResult {
             VerifiableCredentialSdk.correlationVectorService.startNewFlowAndSave()
             val requestResolver = requestResolverFactory.getResolver(verifiedIdRequestInput)
             val rawRequest = requestResolver.resolve(verifiedIdRequestInput)
             val requestHandler = requestHandlerFactory.getHandler(requestResolver)
-            VerifiedIdResult.success(requestHandler.handleRequest(rawRequest))
-        } catch (exception: Exception) {
-            UnspecifiedVerifiedIdException("Unspecified Exception", VerifiedIdExceptions.UNSPECIFIED_EXCEPTION.value, exception).toVerifiedIdResult()
+            requestHandler.handleRequest(rawRequest)
         }
     }
 
     fun encode(verifiedId: VerifiedId): VerifiedIdResult<String> {
         return try {
             VerifiedIdResult.success(serializer.encodeToString(verifiedId))
         } catch (exception: Exception) {
-            UnspecifiedVerifiedIdException("Unspecified Exception", VerifiedIdExceptions.UNSPECIFIED_EXCEPTION.value, exception).toVerifiedIdResult()
+            MalformedInputException(
+                "Malformed Input Exception",
```

Comment:
Should we have a message that the VerifiedId was not able to be encoded?, Also there is parity missing here. We are encoding the Verified Id to Data in iOS, not a String. Let's discuss.

Reply:
Let's talk about this.